과제 1 - Counter 앱 만들고 파일 분리하기

solmatches opened the pull request:
안녕하세요! 2주차 과제를 진행하였습니다 :)

solmatches committed:
feat: Counter 앱 만들기

hannut91 commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,14 @@
+
+function Button({ onClick, children }) {
+  return (
+    <button
+      type="button"
+      onClick={() => onClick(children)}
+    >
+      {children}
+    </button>
```
만약에 보여지는 것과, 우리가 가지고 있는 상태가 다를 경우에는 어떻게 될까요? 예를들어 우리가 가지고 있는 상태는 1, 2, 3 숫자인데, 하나 둘 셋 이렇게 보여주고 싶다고 하면, 이 부분이 어떻게 변경될까요?

hannut91 commented on src/index.jsx:
```diff
@@ -0,0 +1,28 @@
+    <div>
+      <h3>Counter</h3>
+      <Counter count={count} onClick={handleCounterClick} />
+      <Buttons labels={BUTTON_LIST} onClick={handleButtonClick} />
```
나름대로 널리 쓰이는 의미가 있는 단어를 다른 의미로 사용하면 안됩니다. List라는 이름은 프로그래머에게 자료형 List라는 특수한 의미를 가지고 있습니다.  따라서 변수가 실제로 List가 아니라면 변수 이름에 List가 있어서는 안됩니다. 여러개를 나타낸다면 단순히 복수형을 사용하는게 좋습니다.

See also

- 클린코드 - 2장 의미 있는 이름 - 그릇된 정보를 피하라 - [http://www.kyobobook.co.kr/product/detailViewKor.laf?barcode=9788966260959](http://www.kyobobook.co.kr/product/detailViewKor.laf?ejkGb=KOR&mallGb=KOR&barcode=9788966260959&orderClick=LAG&Kc=)

solmatches committed:
fix: BUTTON_LIST -> BUTTON_LABELS 변수이름 변경

실제로 list 자료형이 아닌 변수명에 list를 사용하면 안된다.

보편적으로 사용하는 단어의 의미와 다른 의미로 사용하는 것을 주의

solmatches committed:
refactor: 버튼 표기를 한글로 변경

solmatches commented:
안녕하세요 윤석님! 주신 리뷰를 반영하였습니다! 변수명을 지을 때 습관적으로 list를 많이 사용했었는데 변수 의미에 대해 생각해보게 해주셔서 감사합니다!

hannut91 commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,14 @@
+  );
+}
+
+export default Button;
```
`export default`를 따로 써주신 이유가 있을까요?

solmatches replied:
습관처럼 따로 작성해왔는데 생각해보니 이유가 떠오르지 않네요^^; `export default Button` 과 같이 사용하면 export 되는 것이 어떤 것인지 더 알아차리기 쉬울 것 같다는 생각이 듭니다! 혹시 제가 생각치 못한 이유가 있을까요?

hannut91 replied:
이렇게 쓰는게 컨벤션이 된 이유는 이 사이에 prop types를 쓰거나, redux가 예전에는 `connect`같은 것을 사용해서 중간에 들어갈 코드들이 많아서 썼었는데, TypeScript를 쓰고, `redux`로 `connect`를 쓰지 않다보니 이 사이에 위치할 코드들이 별로 없는 것 같아요. 

따라서 `exprot default function Something() {}`으로 써도 괜찮을 것 같아요. 오히려 모두 `export default`를 구분하는 것보다 중간에 하는 일이 있는 함수만 이렇게 작성하면, 여기서 중간에 하는 일이 있나? 더 눈여겨 볼 것 같아서 눈에 띌 것 같아요.

hannut91 commented on src/components/Buttons.jsx:
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import Button from './Button';
```
위에 빈줄 하나를 추가해서 라이브러리 임포트와 로컬 컴포넌트를 구분해 주세요

hannut91 commented on src/components/Button.jsx:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <button
+      type="button"
+      onClick={() => onClick(children)}
```
여기는 
```js
      onClick={onClick}
```

이렇게 써야겠네요

solmatches committed:
fix: Button 컴포넌트 onClick 수정

Buttons 컴포넌트가 리팩토링되면서 Button 컴포넌트에서 onClick에 children 값을 전달하지 않아도 됨

solmatches committed:
fix: 라이브러리와 로컬 컴포넌트 import 사이에 빈 줄 추가

hannut91 commented on src/index.jsx:
```diff
@@ -0,0 +1,28 @@
+
+const BUTTON_LABELS = [1, 2, 3, 4, 5];
+
+function App() {
```
App 컴포넌트도 다른 파일로 구분하면 좋을 것 같네요

solmatches committed:
refactor: App 컴포넌트 분리 및 export default 위치 수정

- export default 는 HOC 등 중간에 기능이 추가되는 경우에만 분리해서 써준다면
컴포넌트에 붙여서 사용할 때와 분리해서 사용할 때가 확실히 구분되어서 좋을 것 같다!

- App도 컴포넌트 이므로 index에서 분리

hannut91 commented:
회사 일이 너무 바쁘시군요 ㅠㅜ... 조금씩이라도 좋으니 과제를 진행해 보세요. 과제를 하려는 시도만으로도 많은 것을 배우실 수 있으실 거에요.
이번 한 주도 고생 많으셨습니다
